src/nbl/video/IOpenGL_PhysicalDeviceBase.h

```diff
@@ -560,12 +560,88 @@ class IOpenGL_PhysicalDeviceBase : public IOpenGLPhysicalDeviceBase
 			m_properties.limits.maxSamplerAnisotropyLog2 = std::log2((float)m_glfeatures.MaxAnisotropy);
 		}
 		else m_glfeatures.MaxAnisotropy = 0u;
+		
+		GLint64 maxElementIndex = 0;
+		GetInteger64v(GL_MAX_ELEMENT_INDEX, &maxElementIndex);
+		 
+		m_features.fullDrawIndexUint32 = (maxElementIndex == 0xffff'ffff);
+		m_features.independentBlend = (!IsGLES); // TODO
+
+		if (!IsGLES || m_glfeatures.Version >= 320u)
+		{
+			#define GLENUM_WITH_SUFFIX(X) X
+			#include "nbl/video/GL/limit_queries/tessellation_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_OES_tessellation_shader))
+		{
+			#define GLENUM_WITH_SUFFIX(X) X##_OES
+			#include "nbl/video/GL/limit_queries/tessellation_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_EXT_tessellation_shader))
+		{
+			#define GLENUM_WITH_SUFFIX(X) X##_EXT
+			#include "nbl/video/GL/limit_queries/tessellation_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		// else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_INTEL_tessellation_shader))
+		{
+			//!! no _INTEL suffix
+			#define GLENUM_WITH_SUFFIX(X) X##_INTEL
+			// #include "src/nbl/video/GL/limit_queries/tessellation_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		
+		if (!IsGLES || m_glfeatures.Version >= 320u)
+		{
+			#define GLENUM_WITH_SUFFIX(X) X
+			#include "nbl/video/GL/limit_queries/geometry_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_OES_geometry_shader))
+		{
+			#define GLENUM_WITH_SUFFIX(X) X##_OES
+			#include "nbl/video/GL/limit_queries/geometry_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_EXT_geometry_shader))
+		{
+			#define GLENUM_WITH_SUFFIX(X) X##_EXT
+			#include "nbl/video/GL/limit_queries/geometry_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+		// else if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_INTEL_geometry_shader))
+		{
+			//!! no _INTEL suffix
+			#define GLENUM_WITH_SUFFIX(X) X##_INTEL
+			// #include "nbl/video/GL/limit_queries/geometry_shader.h"
+			#undef GLENUM_WITH_SUFFIX
+		}
+
+		m_features.drawIndirectFirstInstance = false; // TODO
+		m_features.depthClamp = m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_ARB_depth_clamp);
```

Comment:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/357#discussion_r889508259

Reply:
its `IsGLES ? (EXT):true`

EXT not ARB